### PR TITLE
fix(isracard): use the correct value for `originalCurrency`

### PR DESCRIPTION
**Pull request by daniel-hauser**
Transactions in isracard has the wrong `originalCurrency`.
The correct mapping of the currencies should be:
- originalCurrency = currentPaymentCurrency
- chargedCurrency = currencyId

------
Attached is an example transaction that should have `originalCurrency: "USD", chargedCurrency: "ILS"` but before the code in this PR it's `originalCurrency: "ILS"` with no `chargedCurrency`

The isracard transaction object:
```
{
    ...
    "currencyId": "NIS",
    "currentPaymentCurrency": "USD",
    ...
    "fullSupplierNameOutbound": "UBER   TRIP",
    ...
    "dealSumOutbound": "66.66",
    "paymentSumOutbound": "248.24",
    ...
}
```

The israeli-bank-scrapers transaction object (diff before and after the change):
```diff
{
         ...
	"originalAmount": -66.66,
-	"originalCurrency": "ILS",
+	"originalCurrency": "USD",
+	"chargedCurrency": "ILS",
	"chargedAmount": -248.24,
	"description": "UBER   TRIP",
	...
}
```

### CODE REVIEW

**Comment by daniel-hauser** [comment]
@baruchiro / @eshaham / @esakal Can you please review? Thanks!

**Comment by baruchiro** [comment]
I know it sounds lazy, but I feel I need more focus to understand it (because I'm taking these reviews sometimes from my phone between things).

Can you edit the description with a "before and after" example, so the change will be thrown דוך to my eyes?

**Comment by daniel-hauser** [comment]
> I know it sounds lazy, but I feel I need more focus to understand it (because I'm taking these reviews sometimes from my phone between things).
> 
> Can you edit the description with a "before and after" example, so the change will be thrown דוך to my eyes?

Done

**Comment by baruchiro** [comment]
@galbarm @daniel-hauser Can you check if in your Dollar transactions there is a mistake?

I'm not paying with my Isracard on Dollars.

**Review by baruchiro — APPROVED** [review]
It's been approved, but I prefer someone who will reproduce this issue and approve your fix.

**Comment by esakal** [comment]
@daniel-hauser  I"m not sure if this is ok or not

CleanShot 2024-07-11 at 08 25 23

the amount is in $ but the currency is nis. this is the outcome of the test 
CleanShot 2024-07-11 at 08 27 37

is this what you expected?

**Comment by daniel-hauser** [comment]
@esakal Can you share the following values from your transaction (in the isracard website)?
```
{
    ...
    "currencyId": ,
    "currentPaymentCurrency": ,
    ...
    "fullSupplierNameOutbound": ,
    ...
    "dealSumOutbound": ,
    "paymentSumOutbound": ,
    ...
}
```

File: src/scrapers/base-isracard-amex.ts [hunk 1]
```diff
@@ -187,8 +188,9 @@ function convertTransactions(txns: ScrapedTransaction[], processedDate: string):
       date: txnMoment.toISOString(),
       processedDate: currentProcessedDate,
       originalAmount: isOutbound ? -txn.dealSumOutbound : -txn.dealSum,
-      originalCurrency: convertCurrency(txn.currencyId),
+      originalCurrency: convertCurrency(txn.currentPaymentCurrency),
```
Comment (daniel-hauser) [inline src/scrapers/base-isracard-amex.ts:191]: Fallback to `currencyId` if `currentPaymentCurrency` is nullish

```suggestion
      originalCurrency: convertCurrency(txn.currentPaymentCurrency ?? txn.currencyId),
```

**Pull request merged**